gtk4/src/custom_filter.rs

```diff
@@ -0,0 +1,80 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::CustomFilter;
+use glib::object::IsA;
+use glib::translate::*;
+use std::ptr;
+
+impl CustomFilter {
+    #[doc(alias = "gtk_custom_filter_new")]
+    pub fn new<F>(filter_func: F) -> CustomFilter
+    where
+        F: Fn(&glib::Object) -> bool + 'static,
+    {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gtk_custom_filter_new(
+                Some(trampoline::<F>),
+                Box::into_raw(Box::new(filter_func)) as *mut _,
+                Some(destroy_closure::<F>),
+            ))
+        }
+    }
+}
+
+impl Default for CustomFilter {
+    fn default() -> Self {
+        assert_initialized_main_thread!();
+        unsafe { from_glib_full(ffi::gtk_custom_filter_new(None, ptr::null_mut(), None)) }
+    }
+}
+
+pub trait CustomFilterExtManual: 'static {
```

Comment:
Just to be sure, this is not a final type and can be subclassed? @bilelmoussaoui?

Otherwise there should be no trait and all functions should be on the type itself.

Reply:
It's a final type, you can subclass a Filter or a Sorter for that purpose.